### DISCUSSION TITLE
Padding for the secure message

### DISCUSSION THREAD

**Discussion by ssunder-xilinx** [DMTF/libspdm#2013] [Q&A]
What if the hardware crypto engine (AES-GCM) on the responder side has a requirement for data length alignment up to a certain number of bytes for crypto operations?  
I don't see the libspdm handling the cipher text's padding to meet the above requirement. The earlier version of the libspdm encode function (`libspdm_encode_secured_message()`) had something related to padding but it has been removed from the latest version. 

Do you think the libspdm should have capability related to the ciphertext alignment requirement of the responder?

**Comment by jyao1** [comment]
AES-GCM does not need block size alignment. It was copy-paste code issue from AES-CBC before.

Why a crypto engine for AES-GCM requires it?
